packages/mui-material/src/Grid/Grid.js

```diff
@@ -29,71 +29,80 @@ function getOffset(val) {
   return `${parse}${String(val).replace(String(parse), '') || 'px'}`;
 }
 
-function generateGrid(globalStyles, theme, breakpoint, ownerState) {
-  const size = ownerState[breakpoint];
-
-  if (!size) {
-    return;
-  }
-
-  let styles = {};
-
-  if (size === true) {
-    // For the auto layouting
-    styles = {
-      flexBasis: 0,
-      flexGrow: 1,
-      maxWidth: '100%',
-    };
-  } else if (size === 'auto') {
-    styles = {
-      flexBasis: 'auto',
-      flexGrow: 0,
-      flexShrink: 0,
-      maxWidth: 'none',
-      width: 'auto',
-    };
-  } else {
-    const columnsBreakpointValues = resolveBreakpointValues({
-      values: ownerState.columns,
-      breakpoints: theme.breakpoints.values,
-    });
-
-    const columnValue =
-      typeof columnsBreakpointValues === 'object'
-        ? columnsBreakpointValues[breakpoint]
-        : columnsBreakpointValues;
-    // Keep 7 significant numbers.
-    const width = `${Math.round((size / columnValue) * 10e7) / 10e5}%`;
-    let more = {};
+function generateGrid({ theme, ownerState }) {
```

Comment:
So the first big change was to convert this method to match `generateDirection` and the other generate methods.  This allowed me to remember the size from the previously defined breakpoint.  So if `xs={1}` is specified and not other breakpoints, each breakpoint will get size 1.  By doing it here, it won't affect ownerState where we would want `sm=false` if sm is not specified.